index.src.html

```diff
@@ -546,25 +519,20 @@ <h4 id="get-types">
   1.  If |response| does not contain a <a>`Clear-Site-Data`</a> header, return
       an empty list.
 
-  2.  Let |types| be an empty list.
-
-  3.  Let |list| be the result of executing the algorithm defined in Section 4
-      of [[!HTTP-JFV]] on the value of |response|'s <a>`Clear-Site-Data`</a>
-      header. If that algorithm results in an error, return an empty list.
-
-  4.  For each |item| in |list|:
+  2.  Let |recognizedTypes| be an empty list.
 
-      1.  If |item| does not have a <a>`types`</a> member, skip to the next
-          |item|.
+  3.  If the value of the <a>`Clear-Site-Data`</a> header is a valid
+      <a grammar>clear-site-data-value</a>, let |types| be the list of all
+      occurences of <a grammar>type</a> in it. Otherwise, return an empty list.
 
-      2.  For each |type| in |item|'s <a>`types`</a> member's value:
+  4.  For each |type| in |types|:
 
-          1.  If |type| is <a>`cache`</a>, <a>`cookies`</a>, <a>`storage`</a>,
-              or <a>`executionContexts`</a>, append |type| to |types|.
+        1.  If |type| is <a>`cache`</a>, <a>`cookies`</a>, <a>`storage`</a>,
+            or <a>`executionContexts`</a>, append |type| to |recognizedTypes|.
 
-              Otherwise, skip to the next |type|.
+            Otherwise, skip to the next |type|.
 
-  5.  Return |types|.
+  5.  Return |recognizedTypes|.
```

Comment:
> Done. Agree with returning failure in level 3, this is what Chromium's implementation actually does (we print an error to the console, we don't just ignore it).

I'd actually suggest changing this to something like "```If |header| is `null` or failure, return an empty list.```", and adding a note that user agents should inform developers of failure to parse the header. If you "return failure", then you need to handle "failure" in https://w3c.github.io/webappsec-clear-site-data/#clear-response instead. I think it makes more sense to handle it here by returning an empty list, which means the clearing operation will just do the right thing (nothing).